document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -577,35 +588,35 @@ public boolean bulkUpsert(Map<Key, Document> documents) {
   public CloseableIterator<Document> bulkUpsertAndReturnOlderDocuments(Map<Key, Document> documents)
       throws IOException {
     String query = null;
-    try {
-      String collect =
-          documents.keySet().stream()
-              .map(val -> "'" + val.toString() + "'")
-              .collect(Collectors.joining(", "));
-
-      String space = " ";
-      query =
-          new StringBuilder("SELECT * FROM")
-              .append(space)
-              .append(collectionName)
-              .append(" WHERE ")
-              .append(ID)
-              .append(" IN ")
-              .append("(")
-              .append(collect)
-              .append(")")
-              .toString();
-
-      PreparedStatement preparedStatement = client.prepareStatement(query);
-      ResultSet resultSet = preparedStatement.executeQuery();
+    String collect =
+        documents.keySet().stream()
+            .map(val -> "'" + val.toString() + "'")
+            .collect(Collectors.joining(", "));
+
+    String space = " ";
+    query =
+        new StringBuilder("SELECT * FROM")
+            .append(space)
+            .append(collectionName)
+            .append(" WHERE ")
+            .append(ID)
+            .append(" IN ")
+            .append("(")
+            .append(collect)
+            .append(")")
+            .toString();
+
+    try (Connection connection = client.getConnection();
```

Comment:
This should not be with `try (Connection connection = client.getConnection();)` as it will close the connection.

Reply:
Fixed this.